cpp/src/parquet/arrow/reader.cc

```diff
@@ -1137,15 +1176,6 @@ struct TransferFunctor<
   Status operator()(RecordReader* reader, MemoryPool* pool,
                     const std::shared_ptr<::arrow::DataType>& type, Datum* out) {
     std::vector<std::shared_ptr<Array>> chunks = reader->GetBuilderChunks();
-
-    if (type->id() == ::arrow::Type::STRING) {
-      // Convert from BINARY type to STRING
-      for (size_t i = 0; i < chunks.size(); ++i) {
-        auto new_data = chunks[i]->data()->Copy();
-        new_data->type = type;
-        chunks[i] = ::arrow::MakeArray(new_data);
-      }
-    }
```

Comment:
Was this logic moved somewhere else, or is it simply not needed?

Reply:
I believe this is no longer needed as the `StringDictionaryBuilder` overloads on the decoders take care of this directly.